### ISSUE TITLE
I have 5 Roku tvs....

### ISSUE THREAD

**Issue by sneakypete419** [FUNKYJASPER2/StreamDeck-Roku-Remote-Plugin#1]
The ip address changes with every instance I put in to the latest ip address. Therefore  i cannot control more than 1 tv. I even tried to use seprate profiles and it would still put in the same ip I last put in. 

Thank you for your work! This is a cool tool that defenitly come in handy!

**Comment by FUNKYJASPER2** [comment]
I can definitely look into adding multiple Roku TV support.

**Comment by BierCoder** [comment]
I'm having the same issue. Thanks for looking into it.

**Comment by timtucker-dte** [comment]
Same here as well -- I have 3 TVs used as monitors and was looking for a good way to turn them all on at once.

**Comment by paddy2me** [comment]
Also have multiple Roku devices in my office and around the house. I would love to be able to control them from one StreamDeck without having to reset the IP address all the time.

**Comment by mangus580** [comment]
I will jump on here too!  Sure would make an easy way of turning off the TV the kids leave on all the time LOL

**Comment by jveeprod** [comment]
I have the same problem. Multiple Roku TV's in my studio. I'd like to set the IP address independently in each StreamDeck button so that I can choose which device to control.

**Comment by jveeprod** [comment]
@FUNKYJASPER2, thanks so much for designing this plugin. 

What are the chances that hardcoding the IP or being able to control multiple devices will be solved?

**Comment by timtucker-dte** [comment]
As FYI, here's the solution I wound up having to use instead -- I just run variations on the following PowerShell script via another Stream Deck plugin:

`# .Net methods for hiding/showing the console in the background
Add-Type -Name Window -Namespace Console -MemberDefinition '
[DllImport("Kernel32.dll")]
public static extern IntPtr GetConsoleWindow();

[DllImport("user32.dll")]
public static extern bool ShowWindow(IntPtr hWnd, Int32 nCmdShow);
'
function Hide-Console
{
    $consolePtr = [Console.Window]::GetConsoleWindow()
    #0 hide
    [Console.Window]::ShowWindow($consolePtr, 0)
}
Hide-Console

# Add the list of IPs for your monitors here
$monitors = '192.168.10.40', '192.168.10.41', '192.168.10.42'

# Run in parallel for faster execution
$monitors | ForEach-Object -Parallel {
	$urlBase = "http://" + $_ + ":8060"
	$keyPress = $urlBase + "/keypress/";
	Invoke-WebRequest -Uri ($keyPress + "PowerOn") -Method POST

	# For power off:
	#Invoke-WebRequest -Uri ($keyPress + "PowerOff") -Method POST
}
`

**Comment by calebvillers** [comment]
I am just getting started using this plug in and am wondering if any progress was made on multiple tv control. I am trying to use it for a Youth Room at our Church that has multiple tvs, and would like to control them all from one location.

**Comment by calebvillers** [comment]
> As FYI, here's the solution I wound up having to use instead -- I just run variations on the following PowerShell script via another Stream Deck plugin:
> 
> [TurnOnMonitors.txt](https://github.com/FUNKYJASPER2/StreamDeck-Roku-Remote-Plugin/files/13490833/TurnOnMonitors.txt)

@timtucker-dte can you expand on how to setup this process?

**Comment by jveeprod** [comment]
@calebvillers No multiple TV control. This has been a request over the last few years but the developer has not replied and seemed to have abandoned the project. It's great if you have one TV but not suitable for multiple TV control as there is no way to hard code the IP address.Thank you for sharing your alternate plugin tip.

**Comment by ChrisOgden** [comment]
@calebvillers I have this working with multiple TVs myself. I need to look at what I did to accomplish it though. I am a developer though and I think I may have downloaded the source, modified it and am running that version instead. Depending on how DIY you want to me I can probably provide that code and you just need to drop the plugin in manually.

**Comment by calebvillers** [comment]
I admit I am not a developer, and so I am rudimentary at best in how to utilize code within projects, but if I can be walked through it that would be great!

**Comment by calebvillers** [comment]
Any update on someone who might be able to share the modified code, and how can I utilize that code on my stream deck?

**Comment by VynashDawnheart** [comment]
Hello, Jasper! Also chiming in on this. I am a now exclusive Roku house, and would love the option of turning off all the TVs in my house, from my desk, so I do not have to go and hunt for remotes when my children and I, inevitably, lose them.

If no update is in the works, that's completely okay. But, could we get some sort of update on the yes/no of it? I couldn't program my way out of a paper bag, so I'm not able to build this out myself, though I wish I could.

Honestly, thank you for what you have done, it's great! Just hoping for multi device support is all. 👍

**Comment by FUNKYJASPER2** [comment]
Thank you all for using the plugin and for your patience. I originally built this when I only had a single Roku TV, so I never prioritized multi-device support. A while ago I sold that TV and since then I have not had any Roku devices, which is why the plugin has not received much attention from me.

I have noticed a steady increase in requests for multi-Roku support recently, and I want to give an update on it.

Current status
	•	I rewrote the plugin at one point to use the newer Node.js SDK that Stream Deck requires, but that was a straightforward port and did not require real Roku testing. I only verified that curl commands were sent.
	•	I do technically have a multi-device version written, but without hardware I have no way to test it.
	•	To move this forward, I plan to purchase two Roku Streaming Sticks so I can test properly against multiple devices.

Why there is no ETA yet
	•	Since I do not currently own any Roku devices, I cannot provide a timeline. Once I have some time and some devices and can test, I will have a better idea of how quickly I can finish and release it.
	•	The Stream Deck team has also confirmed that they are working on plugin DRM protection to reduce third-party stores from redistributing plugins without permission. I would prefer to wait until that protection is active before publishing a new release.

Moving forward

In short:
	•	Multi-Roku support will be added.
	•	I cannot promise an ETA until I have the Roku devices and time to test.
	•	I will post another update once I have devices and progress to share.

Thanks again for the interest and for waiting.

**Comment by ChrisOgden** [comment]
@FUNKYJASPER2 See the pull request I just opened. I thought I had pushed that earlier this year when working on it. I have been using that version for months. I tested it again before committing the code and can control a number of TVs.

@calebvillers I have a forked branch that you could download and drop my version of the files into the plugins folder but hopefully we can just get the PR merged and a new release can be published.

**Comment by iMrAdam** [comment]
@FUNKYJASPER2 
Dude, first this Plugin is so great!  
I would love to send you a couple Roku Express Devices direct from Amazon if it helps Nudge this along.  No strings attached just a way I can say thanks and contribute and maybe see this functionality across the finish line.  Only thing this wont let you test is Power cycle i believe.   Either way THANK YOU!

**Comment by FUNKYJASPER2** [comment]
Hey guys just a small update, I decided to create a discord server for the plugin to make it easier to help you guys out, plus I am opening up a few spots for Beta Testers as the plugin is complete but I need help with testing as my Roku device does not support all features.

https://discord.gg/fZbraQypME
